src/deploy/functions/build.ts

```diff
@@ -586,3 +595,32 @@ function discoverTrigger(endpoint: Endpoint, region: string, r: Resolver): backe
   }
   assertExhaustive(endpoint);
 }
+
+function interpolateEventArcChannel(channel: string, projectId: string): string {
+  if (!channel.includes("/")) {
+    const location = api.functionsDefaultRegion;
+    const channelId = channel;
+    return "projects/" + projectId + "/locations/" + location + "/channels/" + channelId;
+  }
+
+  const CHANNEL_NAME_REGEX = new RegExp(
+    "(projects\\/" +
+      "(?<project>(?:\\d+)|(?:[A-Za-z]+[A-Za-z\\d-]*[A-Za-z\\d]?))\\/)?" +
+      "locations\\/" +
+      "(?<location>[A-Za-z\\d\\-_]+)\\/" +
+      "channels\\/" +
+      "(?<channel>[A-Za-z\\d\\-_]+)"
+  );
+  const match = CHANNEL_NAME_REGEX.exec(channel);
+  if (!match?.groups) {
+    throw new FirebaseError("Invalid channel name format: " + channel);
+  }
+  const matchedProjectId = match.groups.project;
+  const location = match.groups.location;
+  const channelId = match.groups.channel;
+  if (matchedProjectId) {
+    return "projects/" + matchedProjectId + "/locations/" + location + "/channels/" + channelId;
```

Comment:
I think we shouldn't support cross-project channels. IIUC, eventarc doesn't support such a thing anyway no?

I understand this is mostly copy-and-paste code... but just checking to see if we know the answer to issues like cross-region triggering or cross-project triggering. If we don't know the answers, then we should investigate and maybe add some documentation to it at https://firebase.google.com/docs/functions/beta/custom-events.